src/transformStateWithClones.js

```
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  const stateCopy = { ...state };
+
+  for (const step of actions) {
+    switch (step.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, step.extraData);
+        resultArray.push({ ...stateCopy });
+        break;
+      case 'removeProperties':
+        for (const key of step.keysToRemove) {
+          delete stateCopy[key];
+        }
+        resultArray.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        for (const stateKey in stateCopy) {
+          delete stateCopy[stateKey];
+        }
+        resultArray.push({ ...stateCopy });
+        break;
+
+      default:
+        return 'Error';
```

it's better to throw an Error with text like "unexpected action"